docker/buildDocker.sh

```diff
@@ -43,135 +64,103 @@ parseCommandLineArgs()
-		"jdk12u" | "jdk12" | "12" | "12u" )
-			jdkVersion="jdk12u";;
-		"jdk13u" | "jdk13" | "13" | "13u" )
-			jdkVersion="jdk13u";;
-                "jdk14u" | "jdk14" | "14" | "14u" )
-                        jdkVersion="jdk14";;
-                "jdk" | "jdknext" )
-                        jdkVersion="jdk";;
-		"all" )
-			jdkVersion="jdk8u jdk9u jdk10u jdk11u jdk12u jdk13u jdk14u jdk";;
-		*)
-			echo "Not a valid JDK Version" ; jdkVersionList; exit 1;;
-	esac
-}
-
-jdkVersionList()
-{
-	echo
-	echo "Valid JDK versions :
-		- jdk8u
-		- jdk9u
-		- jdk10u
-		- jdk11u
-		- jdk12u
-		- jdk13u
-		- jdk14u
-		- jdk"
+checkJDK() {
+  if ! ((JDK_VERSION >= 8 && JDK_VERSION <= JDK_MAX)); then
+    echo "Please input a JDK between 8 & ${JDK_MAX}, or 'jdk'"
+    echo "i.e. The following formats will work for jdk8: 'jdk8u', 'jdk8' , '8'"
+    exit 1
+  fi
 }
 
 checkArgs()
 {
-	# Sets WORKSPACE to home if WORKSPACE is empty or unbounded
-	if [ ! -n "${WORKSPACE:-}" ]; then
-        	echo "WORKSPACE not found, setting it as environment variable 'HOME'"
-        	WORKSPACE=$HOME
+	# ${WORKSPACE##*/} returns the name of the current dir
+	if [ ${WORKSPACE##*/} != "docker" ]; then
+		echo "Unable to run script from : $WORKSPACE"
+		echo "The script must be run from openjdk-build/docker/"
+		exit 1
 	fi
-	if [ "$cleanWorkspace" == true ]; then
-		echo "Cleaning workspace"
-		rm -rf $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant
+	if [ "$CLEAN" == true ]; then
+		echo "Removing all jdkXX folders"
+		rm -rf $WORKSPACE/jdk*
+		echo "Removing old workspace folder"
+		rm -rf $WORKSPACE/workspace
+		echo "Removing old EclipseDockerfiles folder"
+		rm -rf $WORKSPACE/EclipseDockerfiles
 	fi
 }
 
 useEclipseOpenJ9DockerFiles()
 {
-	mkdir -p $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles
-	cd $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles
+	local dockerfileDir="$WORKSPACE/EclipseDockerfiles"
+	local jdk="jdk"
+
+	mkdir -p $dockerfileDir
+	cd $dockerfileDir
 	wget https://raw.githubusercontent.com/eclipse/openj9/master/buildenv/docker/mkdocker.sh
 	chmod +x mkdocker.sh
 	# Generate an Ubuntu1804 Dockerfile using mkdocker.sh
-	$WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles/mkdocker.sh --dist=ubuntu --version=18 --print >> $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles/Dockerfile
+	$dockerfileDir/mkdocker.sh --dist=ubuntu --version=18 --print >> $dockerfileDir/Dockerfile
 	
 	# This Dockerfile requires an ssh key, authorized_key and known_hosts file to build
-	ssh-keygen -q -f id_rsa -t rsa -N ''
-	cat id_rsa.pub >> authorized_keys
-	ssh-keyscan github.com >> $PWD/known_hosts
+	ssh-keygen -q -f $dockerfileDir/id_rsa -t rsa -N ''
+	cat id_rsa.pub >> $dockerfileDir/authorized_keys
+	ssh-keyscan github.com >> $dockerfileDir/known_hosts
 
-	# ${jdk%?} will remove the 'u' from 'jdkXXu'.
-	for jdk in $jdkVersion
-	do
-		if [ ${jdk} != "jdk" ]; then
-			jdk=${jdk%?}
```

Comment:
Any reason not to use `${jdk%u}` instead of just stripping anything?

Reply:
That's a bit that's been deleted